jupyterhub/apihandlers/auth.py

```diff
@@ -211,53 +241,66 @@ async def get(self):
 
         uri, http_method, body, headers = self.extract_oauth_params()
         try:
-            scopes, credentials = self.oauth_provider.validate_authorization_request(
+            (
+                role_names,
+                credentials,
+            ) = self.oauth_provider.validate_authorization_request(
                 uri, http_method, body, headers
             )
             credentials = self.add_credentials(credentials)
             client = self.oauth_provider.fetch_by_client_id(credentials['client_id'])
-            allowed = False
-
-            # check for access to target resource
-            if client.spawner:
-                scope_filter = self.get_scope_filter("access:users:servers")
-                allowed = scope_filter(client.spawner, kind='server')
-            elif client.service:
-                scope_filter = self.get_scope_filter("access:services")
-                allowed = scope_filter(client.service, kind='service')
-            else:
-                # client is not associated with a service or spawner.
-                # This shouldn't happen, but it might if this is a stale or forged request
-                # from a service or spawner that's since been deleted
-                self.log.error(
-                    f"OAuth client {client} has no service or spawner, cannot resolve scopes."
-                )
-                raise web.HTTPError(500, "OAuth configuration error")
-
-            if not allowed:
-                self.log.error(
-                    f"User {self.current_user} not allowed to access {client.description}"
-                )
-                raise web.HTTPError(
-                    403, f"You do not have permission to access {client.description}"
-                )
```

Comment:
The tests are failing because I lost this block in my rebase. It should be put back as it was.